docs/bugs.json

```diff
@@ -1,4 +1,17 @@
 [
+    {
+        "uid": "SOL-2023-1",
+        "name": "FullInlinerArgumentOrderWithoutExpressionSplitter",
+        "summary": "Optimizer sequences containing FullInliner step not preceded by ExpressionSplitter do not preserve the evaluation order of arguments of inlined function calls.",
+        "description": "Function call arguments in Yul are evaluated right to left. This order matters when the argument expressions have side-effects, and changing it may change contract behavior. FullInliner is an optimizer step that can replace a function call with the body of that function. The transformation involves assigning argument expressions to temporary variables, which imposes an explicit evaluation order. FullInliner was written with the assumption that the argument expressions have no side-effects, and did not attempt to preserve the usual argument evaluation order. The assumption is true under most circumstances because the default optimization step sequence always contains ExpressionSplitter step, which ensures that the expressions are always either identifiers or literals. It is, however, not guaranteed to be true in general. Version 0.6.7 introduced a setting allowing users to specify an arbitrary optimization step sequence, making it possible for the FullInliner to actually encounter argument expressions with side-effects, which can result in behavior differences between optimized and unoptimized bytecode. Contracts compiled without optimization or with the default optimization sequence are not affected. To trigger the bug the user has to explicitly choose compiler settings than contain a sequence with FullInliner step not preceded by ExpressionSplitter.",
+        "link": "TODO",
```

Comment:
No blog post yet.

Reply:
Not necessarily, but to merge it I'll either have to think what URL will be or omit this for now :)

Reply:
Added a tentative link. We still need to remember to recheck that it matches the actual link before the blog post is published.